### ISSUE TITLE
Validation errors after upgrading pyhanko-certvalidator from 0.12.1 to 0.15.1

### ISSUE THREAD

**Issue by gchehab** [MatthiasValvekens/pyHanko#15]
**Describe the bug**
During validation of a pdf file with three signatures, have met with the following warning and the following error:

```
2021-05-20 09:47:22,762 - pyhanko.sign.diff_analysis - WARNING - Error in diff operation between revision 1 and 2
Traceback (most recent call last):
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\diff_analysis.py", line 2130, in review_file
    diff_result = self.apply(
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\diff_analysis.py", line 2030, in apply
    for level, fu in form_changes:
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\diff_analysis.py", line 1168, in apply
    _compare_dicts(old_acroform, new_acroform, self.ignored_acroform_keys)
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\diff_analysis.py", line 1705, in _compare_dicts
    raise SuspiciousModification(
pyhanko.sign.diff_analysis.SuspiciousModification: Dict keys differ: {'/DA', '/SigFlags'} vs. {'/SigFlags'}.
```

```
2021-05-20 09:47:23,482 - pyhanko.cli - ERROR - Generic processing error.
Traceback (most recent call last):
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\cli.py", line 69, in pyhanko_exception_manager
    yield
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\cli.py", line 437, in validate_signatures
    status_str = _signature_status(
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\cli.py", line 327, in _signature_status
    status = validation.validate_pdf_signature(
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\validation.py", line 1263, in validate_pdf_signature
    status_kwargs = _validate_cms_signature(
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\validation.py", line 178, in _validate_cms_signature
    trusted, revoked, path = status_cls.validate_cert_usage(
  File "C:\Users\guilhermec\source\repos\pyHanko\pyhanko\sign\general.py", line 359, in validate_cert_usage
    path = validator.validate_usage(key_usage=set())
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\__init__.py", line 211, in validate_usage
    self._validate_path()
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\__init__.py", line 139, in _validate_path
    validate_path(self._context, candidate_path, self._params)
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\validate.py", line 70, in validate_path
    return _validate_path(validation_context, path, parameters=parameters)
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\validate.py", line 592, in _validate_path
    _check_revocation(
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\validate.py", line 786, in _check_revocation
    verify_crl(
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\validate.py", line 1500, in verify_crl
    certificate_lists = validation_context.retrieve_crls(cert)
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\context.py", line 465, in retrieve_crls
    certs = crl_client.fetch_certs(
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\pyhanko_certvalidator\crl_client.py", line 125, in fetch_certs
    content_type = response.headers['Content-Type'].strip()
  File "c:\Users\guilhermec\source\repos\pyHanko\env\lib\site-packages\requests\structures.py", line 54, in __getitem__
    return self._store[key.lower()][1]
KeyError: 'content-type'
Error: Generic processing error.
```

**To Reproduce**
`python -mpyhanko sign validate --pretty-print testdata/teste3.pdf
`
The used file is attached

**Expected behavior**
The two first signatures should be valide (if Brazilian CA roots are used), the last one invalid since it used a self-signed certificate.
[teste3.pdf](https://github.com/MatthiasValvekens/pyHanko/files/6516421/teste3.pdf)

**Environment (please complete the following information):**
 - OS: Windows 10, python 3.9.5
  - Version commit fd1af25

**Screenshots**
The same validation using adobe reader
![image](https://user-images.githubusercontent.com/5883941/118990441-45404580-b959-11eb-8b82-7db8e1cd890b.png)

**Comment by MatthiasValvekens** [comment]
Hi, thanks for the report!

The first issue has to do with incremental update validation in pyHanko, and can be solved by whitelisting one more key. Someone else had the same issue; I'll do a bugfix release to correct that particular problem soon. In the meantime, here's how to work around the problem by defining your own diff policy:

> As for the workaround, here's the declaration of the default diff
> analysis policy:
> https://github.com/MatthiasValvekens/pyHanko/blob/fd1af2554491fbcd34dbf8bbc29486e2e9a04bef/pyhanko/sign/diff_analysis.py#L2146.
> You can define a custom diff policy to get around this kind of thing,
> and pass it as the `diff_policy` parameter in various validation methods.
> 
> Start by copying the default rules, since you probably want to preserve
> most of those. The FormUpdatingRule class actually takes another
> parameter called `ignored_acroform_keys` where you can pass in keys that
> are to be ignored in the "strict" part of the comparison comparison
> analysis. Passing in "{'/Fields', '/DA', '/DR'}" should be OK.
> 
> Alternatively, you can cheat and add "/DA" to the list of keys in
> pyhanko.sign.diff_analysis.ACROFORM_EXEMPT_STRICT_COMPARISON at runtime.
> Then it will work automagically with the default diff policy 

The second issue is indeed certvalidator-related. It tries to fetch some related certs from a CRL AIA record, but the server isn't setting the Content-Type header. That's is a bit annoying, because there are multiple ways to encode certs in a situation like this, and ordinarily you'd rely on the Content-Type header to select the correct parser. I suspect that this particular fetch wasn't working all along, and my move from `urllib` to `requests` in my fork of `certvalidator` simply broke the error handling code. I'd guess that the certs that `certvalidator` was trying to fetch were actually already available in the local cache anyway.

I'll look into this one more closely, and do a bugfix release if necessary. Thanks for bringing it to my attention!

**Comment by MatthiasValvekens** [comment]
Hi, can you reproduce the problem with the latest `HEAD` for `pyHanko` and `pyhanko-certvalidator`? Your file passes validation on my machine now (well, the last signer's certificate doesn't jive with pyHanko's default key usage policy, but you can easily change that in the [configuration file](https://pyhanko.readthedocs.io/en/latest/cli-guide/config.html#key-usage-settings)).

If this fixes your issue, then I'll do a bugfix release for `pyhanko-certvalidator`, update the dependency in `pyHanko` itself, and do a bugfix release here as well. :)


PS: If you face similar diff analysis issues in the future, please report them if you can. This aspect of PDF signature validation isn't standardised anywhere (yet), so I'm always on the lookout for interesting corner cases.

**Comment by gchehab** [comment]
Hi Matthias,

It did the trick, the validation occurred as expected, The two firsts were considered valid and the latter invalid.  

Regarding the pdf standard 'openness', I know what you mean. I made an wrapper to enable multithread/multiserver batch OCR scanning (https://github.com/gchehab/ocr-server)  and the many diverse ways an embedded image may be encoded almost drove me nuts, I am quite sure that there are a lot of cases that I failed to address. 

I just got my hands on another PDF that a colleague is working on that gives a type error during its parse while validating, I am not sure, however if the issue is not some error on the PDF file signature itself. I'll open an issue specific for this error.

Thank you for your quick support,
Guilherme
